source/adapters/level_zero/v2/event_pool_cache.cpp

```diff
@@ -17,29 +17,35 @@ event_pool_cache::event_pool_cache(size_t max_devices,
                                    ProviderCreateFunc ProviderCreate)
     : providerCreate(ProviderCreate) {
   pools.resize(max_devices);
+  for (auto &p : pools) {
+    p.resize(2 /* profiling enabled or disabled */);
+  }
 }
 
 event_pool_cache::~event_pool_cache() {}
 
-raii::cache_borrowed_event_pool event_pool_cache::borrow(DeviceId id) {
+raii::cache_borrowed_event_pool
+event_pool_cache::borrow(DeviceId id, bool profilingEnabled) {
   std::unique_lock<ur_mutex> Lock(mutex);
 
   if (id >= pools.size()) {
     return nullptr;
   }
 
-  auto &vec = pools[id];
+  auto &vec = pools[id][int(profilingEnabled)];
```

Comment:
Yeah, in general, I thought about creating some simple abstraction for an unordered_container that can be implemented as a vector but decided to keep it simple for now. I've changed the approach to use a continuous vector until we need/have a better abstraction.

Reply:
My point was both about the container, but more so the internal API. Look at this function:
https://github.com/oneapi-src/unified-runtime/blob/main/source/adapters/level_zero/event.cpp#L1259

I think we should be trying to avoid adding random boolean arguments to functions whenever all over the place (just look at how many things you needed to change to do the current change) we need to add a new type of event.
That's why, instead of just adding a `bool profiling` I'd rather we add `uintxxx_t flags`. It's both more readable (the function doesn't need to accept a sequence of `true, true, false, true`) and will be more extensible in the future. It also is easy to integrate with the way you've now changed the pool container (just shift by some EVENT_VALID_FLAGS value instead of 1).

Reply:
Ok, if we're introducing flags right now I think we can also remove event_type enum to make it more consistent. Done.